### PR TITLE
Optional implementation of Vagrant, ORM (Sequelize), Webpack build changes

### PR DESCRIPTION
**Pull request by joedotjs**
- Vagrant has been implemented into the project structure but is an optional workflow for those unacquainted or on low performance machines.

- Sequelize implemented for database ORM, existing schemas (admin and scholar) translated to models and used in new seed. Existing routes updated to use models for querying.

- Webpack development compilation moved to same server as API. Production build unchanged for eventual deployment.

- Some other minor stuff. 😅

### CODE REVIEW

File: build/webpack.dev.config.js [hunk 1]
```diff
@@ -1,42 +1,34 @@
 import webpack from 'webpack';
-import { resolve } from 'path';
```
Comment (lindslev) [inline build/webpack.dev.config.js:2]: u dont like my padding breh???

File: database/seed.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import db from './';
+import {Promise} from 'sequelize'; // is bluebird
+
+const Scholar = db.model('scholar');
+
+const someScholars = [
+  { name: 'Lindsay Levine', email: 'linds@gmail.com', githubUsername: 'lindslev', rank: 1 },
```
Comment (lindslev) [inline database/seed.js:7]: rank 1 get rekt team

File: build/bootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# Hit up that update, son.
```
Comment (lindslev) [inline build/bootstrap.sh:3]: joe ily pls nvr change

File: build/apply-build-middleware.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+export default app => {
+  const webpackCompiler = webpack(webpackConfig);
+  app.use(devMiddleware(webpackCompiler, {
+    publicPath: webpackConfig.output.publicPath,
+    stats: {
+      colors: true
+    }
+  }));
+  app.use(hotMiddleware(webpackCompiler));
+};
```
Comment (lindslev) [inline build/apply-build-middleware.js:16]: this is rad -takes notes-

**Review by lindslev — APPROVED** [review]
🌟

File: build/bootstrap.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+chsh -s $(which zsh) vagrant
+sed -i '/ZSH_THEME=/c\ZSH_THEME="dstufft"' /home/vagrant/.zshrc
+sed -i "/PROMPT=/c\PROMPT='
+%{$fg[magenta]%}%n%{$reset_color%} at %{$fg[yellow]%}%m%{$reset_color%} in %{$fg_bold[g$
+$(virtualenv_info)$(prompt_char) '" /home/vagrant/.oh-my-zsh/themes/dstufft.zsh-theme
+
+echo "So it seems like things probably broke right above this line (Authentication failure/command not found), but it is actually fine. All done!"
+
+    # ----- Environment and Conveniences -----
+echo "export NODE_ENV=development" >> /home/vagrant/.zshrc
```
Comment (covlllp) [inline build/bootstrap.sh:55]: would it make sense to break this out in a separate bash script for the non-vagrant users? (ie. me) would moving setting the various environment variables to an npm script make sense here?

**Review by covlllp — COMMENTED** [review]
small question re: not using vagrant

File: build/bootstrap.sh (hunk 5 above)
Comment (lindslev) [reply build/bootstrap.sh:55]: probably, but will leave that change for a diff PR so i can merge this for joe

**Pull request merged**